### PR TITLE
Fix HUD so it actually works

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
**Currently, for testing purposes, the game is rigged such that Player takes damage each time it collides with a wall.** This is just to ensure that the health system works as intended.

The health system will work as follows:

- Every 2 unit of Player.Health = 1 full Heart
- Odd units of Player.Health imply half hearts

For example, if the Player has 6 Health, then they have 6 / 2 = 3 full hearts.

If the Player has 7 Health, then they have 7 / 2 = 3 full hearts and one half heart.

Each time the Player's health or max health changes, HeartSystem updates the heart containers on screen accordingly.

### CODE REVIEW

File: Assets/Scripts/Player.cs [hunk 1]
```diff
@@ -4,10 +4,13 @@
 
 public class Player : Entity
 {
+    public delegate void HealthChanged();
+    public static event HealthChanged healthChangedEvent;
```
Comment (Jazy5552) [inline Assets/Scripts/Player.cs]: What is the reason for making this static?

File: Assets/Scripts/Player.cs [hunk 2]
```diff
@@ -22,14 +25,30 @@ public override void Attack()
         // TODO code
     }
 
+    public override void ChangeMaxHealth(int amount)
+    {
+        MaxHealth += amount;
+        healthChangedEvent?.Invoke();
```
Comment (Jazy5552) [inline Assets/Scripts/Player.cs]: Neat null-conditional, didn't even know these were a thing.

File: Assets/Scripts/HeartSystem.cs [hunk 3]
```diff
@@ -17,61 +17,70 @@ public class HeartSystem : MonoBehaviour
     private void Start()
     {
         player = gameObject.GetComponent<Player>();
+        Player.healthChangedEvent += OnPlayerHealthChanged;
```
Comment (Jazy5552) [inline Assets/Scripts/HeartSystem.cs]: Seems like making the `healthChangedEvent` non-static would simplify this, no? You'd be able to use the `player` object you just got right above.

File: Assets/Scripts/Player.cs (hunk 2 above)
Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Player.cs]: Thank Intellisense :)

File: Assets/Scripts/Player.cs (hunk 1 above)
Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Player.cs]: I was mostly just following a tutorial to throw something together that works for event listening/detection. New to C#, so not entirely sure how the static modifier interacts with events :)

Comment (Jazy5552) [reply Assets/Scripts/Player.cs]: Ah ok. I'd think non-static would be more appropriate here. Doesn't really matter just wanted to point it out.

**Pull request merged**